go/libraries/doltcore/table/editor/keyless_table_editor.go

```diff
@@ -493,10 +494,10 @@ func applyEdits(ctx context.Context, tbl *doltdb.Table, acc keylessEditAcc, inde
 			return nil, err
 		}
 
-		func(k, v types.Tuple) (*doltdb.Table, error) {
+		err = func(k, v types.Tuple) (localErr error) {
 			indexOpsToUndo := make([]int, len(indexEds))
 			defer func() {
-				if retErr != nil {
+				if localErr != nil {
```

Comment:
@max-hoffman TBH i still haven't found an edge case that break this code. Would love some insight on why you wrote and whether it should be salvaged or not.